add zlib

# Situation:

The env-builder build is failing at step 34, which are commands related to installing python.
```
Step 34 : RUN VERSION='3.6.1' &&   wget https://www.python.org/ftp/python/${VERSION}/Python-${VERSION}.tgz &&   tar zxvf Python-${VERSION}.tgz &&   cd Python-${VERSION} &&   ./configure &&   make &&   make install &&   cd .. &&   rm -r Python-${VERSION}.tgz Python-${VERSION}
```
fails at line 23196:
```
Traceback (most recent call last):
  File "/root/Python-3.6.1/Lib/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/root/Python-3.6.1/Lib/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/root/Python-3.6.1/Lib/ensurepip/__main__.py", line 4, in <module>
    ensurepip._main()
  File "/root/Python-3.6.1/Lib/ensurepip/__init__.py", line 189, in _main
    default_pip=args.default_pip,
  File "/root/Python-3.6.1/Lib/ensurepip/__init__.py", line 102, in bootstrap
    _run_pip(args + [p[0] for p in _PROJECTS], additional_paths)
  File "/root/Python-3.6.1/Lib/ensurepip/__init__.py", line 27, in _run_pip
    import pip
zipimport.ZipImportError: can't decompress data; zlib not available
make: *** [install] Error 1
The command '/bin/sh -c VERSION='3.6.1' &&   wget https://www.python.org/ftp/python/${VERSION}/Python-${VERSION}.tgz &&   tar zxvf Python-${VERSION}.tgz &&   cd Python-${VERSION} &&   ./configure &&   make &&   make install &&   cd .. &&   rm -r Python-${VERSION}.tgz Python-${VERSION}' returned a non-zero code: 2

docker build -t all all returned exit code 1
```

# Cause Hypothesis
docker build is failing due to zlib missing
line 23209 reads:
```
zipimport.ZipImportError: can't decompress data; zlib not available
```

# Fix Proposal
add zlib with the following command, maybe?:
```
sudo apt-get install zlib1g-dev
```
Ref: https://unix.stackexchange.com/questions/291737/zipimport-zipimporterror-cant-decompress-data-zlib-not-available

--- a/all/Dockerfile
+++ b/all/Dockerfile
@@ -127,8 +127,8 @@ RUN VERSION='2.7.11' && \
   make install && \
   cd .. && \
   rm -r Python-${VERSION}.tgz Python-${VERSION}
-RUN curl --silent https://bootstrap.pypa.io/get-pip.py | python3
-RUN curl --silent https://bootstrap.pypa.io/get-pip.py | python
+# RUN curl --silent https://bootstrap.pypa.io/get-pip.py | python3
+# RUN curl --silent https://bootstrap.pypa.io/get-pip.py | python
 RUN pip install nose
 RUN pip3 install nose
 

--- a/all/Dockerfile
+++ b/all/Dockerfile
@@ -108,7 +108,7 @@ RUN apt-add-repository -y ppa:webupd8team/java && \
 # python
 RUN apt-get update -qq && \
   apt-get install -qqy build-essential libssl-dev libbz2-dev libreadline-dev libsqlite3-dev zlib1g-dev && \
-  export LD_LIBRARY_PATH="/usr/local/ssl/lib" && \
+  export LD_LIBRARY_PATH="/usr/include/openssl/" && \
   rm -rf /var/lib/apt/lists/*
 RUN VERSION='3.6.1' && \
   wget https://www.python.org/ftp/python/${VERSION}/Python-${VERSION}.tgz && \

--- a/all/Dockerfile
+++ b/all/Dockerfile
@@ -108,6 +108,7 @@ RUN apt-add-repository -y ppa:webupd8team/java && \
 # python
 RUN apt-get update -qq && \
   apt-get install -qqy build-essential libssl-dev libbz2-dev libreadline-dev libsqlite3-dev zlib1g-dev && \
+  export LD_LIBRARY_PATH="/usr/local/ssl/lib" && \
   rm -rf /var/lib/apt/lists/*
 RUN VERSION='3.6.1' && \
   wget https://www.python.org/ftp/python/${VERSION}/Python-${VERSION}.tgz && \
@@ -127,8 +128,8 @@ RUN VERSION='2.7.11' && \
   make install && \
   cd .. && \
   rm -r Python-${VERSION}.tgz Python-${VERSION}
-# RUN curl --silent https://bootstrap.pypa.io/get-pip.py | python3
-# RUN curl --silent https://bootstrap.pypa.io/get-pip.py | python
+RUN curl --silent https://bootstrap.pypa.io/get-pip.py | python3
+RUN curl --silent https://bootstrap.pypa.io/get-pip.py | python
 RUN pip install nose
 RUN pip3 install nose
 

--- a/all/Dockerfile
+++ b/all/Dockerfile
@@ -107,7 +107,7 @@ RUN apt-add-repository -y ppa:webupd8team/java && \
 
 # python
 RUN apt-get update -qq && \
-  apt-get install -qqy build-essential libssl-dev libbz2-dev libreadline-dev libsqlite3-dev && \
+  apt-get install -qqy build-essential libssl-dev libbz2-dev libreadline-dev libsqlite3-dev zlib1g && \
   rm -rf /var/lib/apt/lists/*
 RUN VERSION='3.6.1' && \
   wget https://www.python.org/ftp/python/${VERSION}/Python-${VERSION}.tgz && \

--- a/all/Dockerfile
+++ b/all/Dockerfile
@@ -107,7 +107,7 @@ RUN apt-add-repository -y ppa:webupd8team/java && \
 
 # python
 RUN apt-get update -qq && \
-  apt-get install -qqy build-essential libssl-dev libbz2-dev libreadline-dev libsqlite3-dev zlib1g && \
+  apt-get install -qqy build-essential libssl-dev libbz2-dev libreadline-dev libsqlite3-dev zlib1g-dev && \
   rm -rf /var/lib/apt/lists/*
 RUN VERSION='3.6.1' && \
   wget https://www.python.org/ftp/python/${VERSION}/Python-${VERSION}.tgz && \

